src/main/java/com/google/api/codegen/transformer/PageStreamingTransformer.java

```diff
@@ -31,7 +31,7 @@
   public List<PageStreamingDescriptorView> generateDescriptors(SurfaceTransformerContext context) {
     List<PageStreamingDescriptorView> descriptors = new ArrayList<>();
 
-    for (Method method : context.getInterface().getMethods()) {
+    for (Method method : context.getNonStreamingMethods()) {
```

Comment:
I was dubious about putting getNonStreamingMethods in the context, but after some thought, I have decided to embrace the approach. In this particular case, the loop is supposed to be over page streaming methods. So, could you add a getPageStreamingMethods() method in the context, use it here, and remove the isPageStreaming() check below?